### PR TITLE
Prototype passcode change UI

### PR DESCRIPTION
**Pull request by miika22**
I like how this screen looks on Android, but it still needs to be tested on iOS. I still need to add tests and maybe clean up some stuff.

You can test it by setting HomeScreen's switchToSettings function to route to: "PasscodeChange"

### CODE REVIEW

**Comment by richardkriesman** [comment]
[Prototype passcode change UI](https://trello.com/c/XdNoomqS/50-prototype-passcode-change-ui)

**Comment by miika22** [comment]
The request is ready for review. Test file was added earlier today, and I can't think of any changes to make

File: src/utils/AppDatabase.ts [hunk 1]
```diff
@@ -27,14 +33,21 @@ export class AppDatabase {
...
 
+        AppDatabase._initCount++;
         return appDb;
     }
 
+    /**
+     * Tracks the number of database initializations. Used to catch accidental re-initializations when the connection
+     * is not properly passed between screens.
+     */
+    private static _initCount: number = 0;
```
Comment (ChaoticWeg) [inline src/utils/AppDatabase.ts:44]: Are we ever actually using this? Also, what's the advantage of this over a static instance with safe single init?

Comment (ChaoticWeg) [reply src/utils/AppDatabase.ts:44]: Both honest questions, oops.

File: src/utils/Log.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
+/**
```
Comment (ChaoticWeg) [inline src/utils/Log.ts:1]: Sexy.

File: src/utils/TestUtils.ts [hunk 3]
```diff
@@ -2,17 +2,75 @@
...
- * Creates a new mock object for react-navigation's navigation object.
+ * A class containing mocks and utilities needed for running automated tests.
  */
-export function createNavigationMock(): any {
-    return {
-        getParam: jest.fn(),
-        navigation: jest.fn(),
-        setParams: jest.fn()
-    };
+export class TestEnvironment {
```
Comment (ChaoticWeg) [inline src/utils/TestUtils.ts:16]: !!!

**Review by ChaoticWeg — COMMENTED** [review]
Looks good to me. Only that one question with `AppDatabase`, which doesn't necessarily block the review; just looking for feedback on it.

File: src/utils/AppDatabase.ts (hunk 1 above)
Comment (richardkriesman) [reply src/utils/AppDatabase.ts:44]: For the first question: yup! It's being used by `UIScreen` when it calls the `init()`.

For the second question: it keeps `AppDatabase` instances from being stored in global state. I don't see too much of an issue with that in some cases where you need to share a single connection per process, but some people see it as an anti-pattern.

File: src/utils/Log.ts (hunk 2 above)
Comment (richardkriesman) [reply src/utils/Log.ts:1]: I used a similar design in another project of mine and it ended up working really well and really cleaning up logging. Thought it'd be useful here too 😄

**Review by ChaoticWeg — DISMISSED** [review]

File: src/utils/AppDatabase.ts (hunk 1 above)
Comment (ChaoticWeg) [reply src/utils/AppDatabase.ts:44]: Excellent. Thanks!

**Review by richardkriesman — APPROVED** [review]

**Pull request merged**